### PR TITLE
TIP-732: Rework attribute form to new front architecture

### PR DESCRIPTION
**Pull request by phaseinducer**
**Description (for Contributor and Core Developer)**

TODO:

- [x] Change normalizer
- [x] Change routing
- [x] Implement a way to build type-specific form parts
- [x] Implement common fields
- [x] Implement type-specific fields: Date
- [x] Implement type-specific fields: File / Image
- [x] Implement type-specific fields: Identifier
- [x] Implement type-specific fields: Metric
- [x] Implement type-specific fields: Number
- [x] Implement type-specific fields: Simple-select
- [x] Implement type-specific fields: Multi-select
- [x] Implement type-specific fields: Ref data
- [x] Implement type-specific fields: Price
- [x] Implement type-specific fields: Text
- [x] Implement type-specific fields: Textarea
- [x] Add form save / validation
- [x] Add read only behaviors
- [x] Add translations tab
- [x] Add history tab
- [x] Add rules tab
- [x] Delete attribute
- [x] Create attribute
- [x] Handle ACLs
- [x] Trash old backend actions, twig templates & form types
- [x] Fix behats
- [ ] List BC breaks

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added Behats                      | -
| Added integration tests           | OK
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: frontend/config-loader.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* eslint-env es6 */
 const utils = require('loader-utils')
 const path = require('path')
-const hasModule = (content) => content.indexOf('module') >= 0
+const hasModule = (content) => content.indexOf('__moduleConfig') >= 0
```
Comment (tamarasaurus) [inline frontend/config-loader.js]: 👍

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute/form/choices.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/**
+ * @author    Yohan Blain <yohan.blain@akeneo.com>
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute/form/choices.js]: Add description mofo

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute/form/properties/field.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+         * @param {*} value
+         */
+        updateModel: function (value) {
+            var newData = {};
+            newData[this.fieldName] = value;
+
+            this.setData(newData);
+        },
+
+        getFieldValue: function () {
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute/form/properties/field.js]: jsdoc

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute/form/properties/field.js [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+         * Should the field be in readonly mode?
+         *
+         * @returns {Boolean}
+         */
+        isReadOnly: function () {
+            return this.config.readOnly || false;
+        },
+
+        /**
+         * @param {*} value
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/attribute/form/properties/field.js]: description

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/add-select/attribute/select.js [hunk 5]
```diff
@@ -60,6 +60,29 @@ define(
...
+             * This method is overridden to fetch attribute groups and set it inside attribute items.
+             *
+             * {@inheritdoc}
+             */
+            fetchItems: function () {
+                return BaseAddSelect.prototype.fetchItems.apply(this, arguments)
+                    .then(function (attributes) {
+                        var groupCodes = _.unique(_.pluck(attributes, 'group'));
+
+                        return FetcherRegistry.getFetcher('attribute-group').fetchByIdentifiers(groupCodes)
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/add-select/attribute/select.js]: move it to a dedicated method mofo

File: src/Pim/Bundle/UIBundle/Resources/public/lib/select2/select2.css [hunk 6]
```diff
@@ -592,7 +592,8 @@ disabled look for disabled choices in the results dropdown
     background-color: #f4f4f4;
 }
 
-.select2-container-multi.select2-container-disabled .select2-choices .select2-search-choice .select2-search-choice-close {    display: none;
+.select2-container-multi.select2-container-disabled .select2-choices .select2-search-choice .select2-search-choice-close {
```
Comment (phaseinducer) [inline src/Pim/Bundle/UIBundle/Resources/public/lib/select2/select2.css]: revert mofo

**Comment by phaseinducer** [comment]
Mob reviewed with @tamarasaurus and @pierallard :dancer:

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/add-select/attribute/select.js (hunk 5 above)
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/add-select/attribute/select.js]: done

**Pull request merged**